Remove CFLAG crap and clean up the --enable-{debug,md5} options 
to work correctly.

--- mbuffer-20050125/configure.in
+++ mbuffer-20050125/configure.in
@@ -23,17 +23,6 @@
 AC_SEARCH_LIBS(sem_init,rt posix,,AC_MSG_ERROR(could not find semaphore functions!))
 AC_PATH_PROG(MT,mt,AC_MSG_WARN(could not find the program mt - you might need this if you want autoloader support))
 
-if test -n "$GCC"; then
-        CFLAGS="$CFLAGS -Wall"
-        case "$target_cpu" in
-                i586*)
-                CFLAGS="$CFLAGS -march=pentium"
-                ;;
-                i686*)
-                CFLAGS="$CFLAGS -march=pentiumpro"
-                ;;
-        esac
-fi
 
 AC_TYPE_SIGNAL
 AC_FUNC_MMAP
@@ -49,15 +38,26 @@
 
 AC_ARG_ENABLE(debug,
         --disable-debug           disable verbose logging to console for debugging purpose,
-        ,AC_DEFINE_UNQUOTED(DEBUG),
+        ,enable_debug=yes
         )
+if test x$enable_debug = xyes ; then
+	AC_DEFINE_UNQUOTED(DEBUG)
+fi
 
 AC_ARG_ENABLE(md5,
         --disable-md5             disable md5 hash generation,
-	,
-	AC_SEARCH_LIBS(mhash_init,mhash,AC_DEFINE_UNQUOTED(MHASH),
-		AC_SEARCH_LIBS(MD5Init,md5,AC_DEFINE_UNQUOTED(MD5),AC_MSG_WARN(could not find the mhash library containing the md5 hash functions - md5 support disabled)))
+	,enable_md5=auto
 	)
+if test x$enable_md5 != xno ; then
+	AC_SEARCH_LIBS(mhash_init,mhash,AC_DEFINE_UNQUOTED(HAVE_MHASH),
+		AC_SEARCH_LIBS(MD5Init,md5,AC_DEFINE_UNQUOTED(MD5),
+		if test x$enable_md5 = xauto ; then
+			AC_MSG_WARN(could not find the mhash library containing the md5 hash functions - md5 support disabled)
+		else
+			AC_MSG_ERROR(could not find the mhash library)
+		fi
+	))
+fi
 
 AC_SUBST(DEBUG)
 AC_DEFINE_UNQUOTED(PACKAGE,"${PACKAGE}")
